src/comment.rs

```diff
@@ -1304,7 +1304,13 @@ impl<'a> Iterator for LineClasses<'a> {
             None => FullCodeCharKind::Normal,
         };
 
-        while let Some((kind, c)) = self.base.next() {
+        while let Some((kind, mut c)) = self.base.next() {
+            // If \r\n newline appears, consume one more character.
+            // Then do the same process with the single \n case.
+            if c == '\r' && self.base.peek().map_or(false, |(_, c2)| *c2 == '\n') {
+                self.base.next();
+                c = '\n';
```

Comment:
I am a bit uncomfortable making this `if` dependent on the next one by changing the value of `c`.

It seems that this `if` could be moved inside the next one: you check if `line` ends with `\r` and call [String::pop](https://doc.rust-lang.org/std/string/struct.String.html#method.pop) on `line` if that is the case.

Does that work ?